pkg/apis/migration/migration.go

```diff
@@ -235,6 +239,38 @@ func (volumeMigration *volumeMigration) GetVolumeID(ctx context.Context, volumeS
 	return "", ErrVolumeIDNotFound
 }
 
+// ProtectVolumeFromVMDeletion sets FCD_KEEP_AFTER_DELETE_VM control flag on the migrated volume
+// Returns an error if not able to set FCD_KEEP_AFTER_DELETE_VM control flag on the volume
+func (volumeMigration *volumeMigration) ProtectVolumeFromVMDeletion(ctx context.Context, volumeID string) error {
+	log := logger.GetLogger(ctx)
+	volumeMigrationResource := &migrationv1alpha1.CnsVSphereVolumeMigration{}
+	var err error
+	err = volumeMigration.k8sClient.Get(ctx, client.ObjectKey{Name: volumeID}, volumeMigrationResource)
+	if err != nil {
+		log.Errorf("error happened while getting CR for volumeMigration for VolumeID: %q, err: %v", volumeID, err)
+		return err
+	}
+	if !volumeMigrationResource.Spec.ProtectVolumeFromVMDelete {
+		log.Infof("Set FCD_KEEP_AFTER_DELETE_VM control flag using Vslm APIs")
+		err = (*volumeMigration.volumeManager).ProtectVolumeFromVMDeletion(ctx, volumeID)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Infof("migrated volume with ID: %q is already protected from vm deletion", volumeID)
+		return nil
+	}
+	log.Infof("Migrated Volume with ID: %q is protected from VM deletion", volumeID)
+	volumeMigrationResource.Spec.ProtectVolumeFromVMDelete = true
+	err = volumeMigration.k8sClient.Update(ctx, volumeMigrationResource)
+	if err != nil {
```

Comment:
How about handling conflict error during update(the version you are trying to update is already old) ?
Asking this because, if it fails here due to conflict error the final error & fault will be reported as `CSIInternalFault` prometheus

Reply:
We are getting the cnsvspherevolumemigrations CR object created by the Syncer after the volume is migrated.
If the Syncer has not migrated volume we will not have CR.
and if CR is present the only update we need to make in the spec is to set `protectvolumefromvmdelete` to true.

also migrated volumes are of RWO type, so don't have multiple `controller publish` calls attempting to update this CR in parallel.

considering the above we are safe.